src/nonlinearshrinkage.jl

```diff
@@ -56,11 +58,11 @@ function analytical_nonlinear_shrinkage(X::AbstractMatrix; decomp::Union{Eigen,N
     U    = F.vectors[:, perm]
 
     # compute analytical nonlinear shrinkage kernel formula
-    λ = λ[max(1, p-n+1):p]
-    L = repeat(λ, outer=(1, min(p, n)))
+    λ = (p < n) ? λ[1:p] : λ[2+(p-n):p]
+    L = repeat(λ, outer=(1, min(p, n - 1))) # size (2+(p-n)) x n
 
     # Equation (4.9)
-    h = n^(-1.0/3.0)
+    h = n^(-1//3)
```

Comment:
Since the effective sample size is `n-1`, shouldn't it be `n-1` here as well?

Reply:
And similarily for `γ  = p/n`.

Reply:
I think I agree for h, and also for the definition of L (repmat min(p, ...)) but not for gamma (otherwise HF0 is problematic because the log gets a negative argument). 
Anyway this stuff is a bit messy... I would have hoped the authors had replied something more explicit.